### PR TITLE
Modularize Post Processing

### PR DESCRIPTION
**Pull request by damonbayer**
This is a draft PR until my requested feature in `renv` is added. It is already in an open PR: https://github.com/rstudio/renv/pull/2018

The `renv` bit is just required for local development (and I could come up with a workaround if required). The modularized R script should already work in an azure context.

Closes https://github.com/CDCgov/pyrenew-hew/issues/40

### CODE REVIEW

File: nssp_demo/post_process.R [hunk 1]
```diff
@@ -5,12 +5,76 @@ library(cowplot)
...
+
+  if (filter_bad_chains) {
+    good_chains <-
+      pyrenew_samples$log_likelihood %>%
+      pivot_longer(-starts_with(".")) %>%
+      group_by(.iteration, .chain) %>%
+      summarize(value = sum(value)) %>%
+      group_by(.chain) %>%
+      summarize(value = mean(value)) %>%
+      filter(value >= max(value) - 2) %>%
```
Comment (dylanhmorris) [inline nssp_demo/post_process.R:62]: `log(target_fold_difference)` for interpretability?

```suggestion
      filter(value >= max(value) - log(5)) %>%
```

File: nssp_demo/post_process.R [hunk 2]
```diff
@@ -118,35 +160,34 @@ make_forecast_fig <- function(model_dir) {
...
 
-base_dir <- path(here(
+save_plot(
+  filename = path(model_dir, "forecast_plot", ext = "pdf"),
+  plot = forecast_fig,
+  device = cairo_pdf, base_height = 6
+)
+
+
+# Temp code while command line version doesn't work
```
Comment (dylanhmorris) [inline nssp_demo/post_process.R:172]: Noting that this can be fixed in #66

Comment (damonbayer) [reply nssp_demo/post_process.R:172]: This is dependent on https://github.com/rstudio/renv/pull/2018.

**Comment by dylanhmorris** [comment]
@damonbayer will let you merge

**Pull request merged**